feat: rooms-screen

haaszdev opened the pull request:
- Tela de salas implementada, porém sujeita a alterações assim que realizarmos a tela de turmas.
- Minha ideia é:
A turma vai puxar o curso, e na turma cadastraremos o período daquela determinada turma, para termos uma identificação, por exemplo:
 - Turma de Engenharia de Software, 4 período, possui X alunos, e isso eventualmente, sera usado para o cadastro da sala, pois na mesma poderemos puxar a turma, exemplo: Sala 308 - Turma Engenharia de Software - 4 periodo.
 
 Com isso, no momento em que fizermos as turmas, ajusto a listagem de Salas.

haaszdev committed:
feat: rooms-screen

Lucas-Dreveck committed:
Merge branch 'development' into feat/rooms-screen

Lucas-Dreveck committed:
fix: resolve eslint warnings and apply code formatting

applied eslint and prettier fixing proper TS types and interfaces. Add sonarqube coverage folder to
eslint ignore

Lucas-Dreveck committed:
chore: return all web dependencies to the development branch versions

For some reason all dependencies of the web `package.json` was in a old version that is the answer
to why the pipeline (and the actual code) was having problems

heloshartmann reviewed (changes requested):
Necessário alumas mudanças:
- Pouco espaço no campo de filtro
<img width="403" height="139" alt="image" src="https://github.com/user-attachments/assets/65c15782-b6e3-4c7a-ab69-460169ff0d0a" />

- Modal de nova sala e editar pequeno
<img width="667" height="594" alt="image" src="https://github.com/user-attachments/assets/df2a6baa-6cad-4ee2-bdd9-54c8f5d3b4b2" />

- Não mostra os cursos realmente cadastrados
<img width="371" height="249" alt="image" src="https://github.com/user-attachments/assets/eff82d5f-d6dc-4425-9e3f-4d157e5d1a57" />

- Erro ao adicionar nova sala
<img width="1906" height="900" alt="image" src="https://github.com/user-attachments/assets/3d217d3a-1377-4c05-9834-36b2e65f3448" />

- No campo de turma deve mostrar o curso
<img width="253" height="361" alt="image" src="https://github.com/user-attachments/assets/ecc80cef-6b16-4c84-b005-d5adfb7d6d90" />

heloshartmann committed:
feat: ordem sidebar

GusDechotti committed:
fix: correções que a helo pediu

heloshartmann reviewed (changes requested):
Adicionar o campo de curso e turma

heloshartmann committed:
Merge branch 'development' into feat/rooms-screen

GusDechotti committed:
fix: adicionada coluna de nome da turma

Lucas-Dreveck committed:
Merge branch 'development' into feat/rooms-screen

Lucas-Dreveck committed:
fix(room-page): added courseId for visualization of partial data

added courseId for visualization of partial data where we don't have a class so it's possible to
declare it in a second step. Plus I fixed the parse of size to be show in the edit modal

Lucas-Dreveck committed:
Merge branch 'development' into feat/rooms-screen

Lucas-Dreveck committed:
test(rooms): updated tests to have the new fields

put the course and courseId fields in the unit tests

Lucas-Dreveck reviewed (approved):
Tela funcionando 100%
